### ISSUE TITLE
Weird terminal display when using your bash profile template

### ISSUE THREAD

**Issue by terichadbourne** [susanBuck/e28-fall19#21]
@susanBuck I tried adding your proposed bash items to the end of my previous file, commenting out the old options that seemed to conflict. After restarting terminal, if I type more than fits on one line, instead of going to the next line it goes to the left side of the current line and types over the prompt, like so: 

![image](https://user-images.githubusercontent.com/19171465/65078691-89e93180-d96b-11e9-9504-49f6a6f0865a.png)

![image](https://user-images.githubusercontent.com/19171465/65078707-93729980-d96b-11e9-8880-9b035b82dcf5.png)

Any idea what might cause this? 
```
# Git

.bashrc looks like this: 
######### used prior to e28 109-09-17  ######
# function parse_git_branch {
#   ref=$(git symbolic-ref HEAD 2> /dev/null) || return
#   echo "("${ref#refs/heads/}")"
# }
# export PS1="\[$(tput bold)\]\w\[$(tput sgr0)\] \$(parse_git_branch)\n$ "
export EDITOR='atom --wait'
export VISUAL='atom --wait'


  export NVM_DIR=~/.nvm
  . $(brew --prefix nvm)/nvm.sh

# Rbenv
if which rbenv > /dev/null; then
   eval "$(rbenv init -)"
fi


###### added for e28 2019-09-17 #####

# If not running interactively, don't do anything
[ -z "$PS1" ] && return


# ------------------------------------
# ALIASES
# ------------------------------------
# Edit .bashrc
alias bashedit='sudo open -a TextEdit ~/.bashrc'

# Force terminal to recognize changes to .bashrc
alias bashrefresh='source ~/.bashrc'

# Ideal directory listing
alias ll="ls -laFG"
# -l list in long format
# -F Display a...
#       slash (`/') immediately after each pathname that is a directory,
#       asterisk (`*') after each that is executable,
#       at sign (`@') after each symbolic link
#       equals sign (`=') after each socket,
#       percent sign (`%') after each whiteout,
#       vertical bar (`|') after each that is a FIFO.
# -G = Color
# -a = Show hidden files

# Ask before removing files
alias rm='rm -i'

# Search history. Example usage: `histg git` to recent commands that use git
alias histg="history | grep"

# Get your current IP
alias myip="curl http://ipecho.net/plain; echo"

# Example alias for SSH'ing into a server
alias myserver="ssh user@111.111.111.111"

# Example alias for quickly getting to a commonly used directory
alias htdocs='cd /Applications/XAMPP/xamppfiles/htdocs'
alas e28='cd /Applications/XAMPP/xamppfiles/htdocs/e28'


# ------------------------------------
# Color variables
# ------------------------------------
# Regular Colors
Black='\e[0;30m'
Red='\e[0;31m'
Green='\e[0;32m'
Yellow='\e[0;33m'
Blue='\e[0;34m'
Purple='\e[0;35m'
Cyan='\e[0;36m'
White='\e[0;37m'
Light_Gray='\033[0;37m'

# Bold
BBlack='\e[1;30m'
BRed='\e[1;31m'
BGreen='\e[1;32m'
BYellow='\e[1;33m'
BBlue='\e[1;34m'
BPurple='\e[1;35m'
BCyan='\e[1;36m'
BWhite='\e[1;37m'
BLight_Gray='\033[1;37m'

# Reset colors
NONE="\e[0m"


# ------------------------------------
# Configure prompt
# Includes special handling for git repos
# ------------------------------------
# When in a git repo, this method is used to determine the current branch
function parse_git_branch {
    git branch 2>/dev/null | grep '^*' | sed 's_^..__' | sed 's_\(.*\)_(\1)_'
}

# When in a git repo, this method is used to determine if there are changes
function git_dirty {
    git diff --quiet HEAD &>/dev/null
    [ $? == 1 ] && echo "!"
}

# Variables
ps1_user="$Blue\u$NONE"
ps1_host="$Green\h$NONE"
ps1_dir="$Purple\w$NONE"
ps1_git="$Cyan \$(parse_git_branch)$Red \$(git_dirty)$NONE "

# Option 1 user@host:dir(branch)! $
# export PS1="${ps1_user}@${ps1_host}:${ps1_dir}${ps1_git}\$ "

# Option 2 dir(branch)! $
#export PS1="${ps1_dir}${ps1_git}\$ "

export PS1="${ps1_dir}${ps1_git}\$ "


# ------------------------------------
# MOTD (Message of the Day)
# What you see when Terminal opens
# ------------------------------------

printf $BBlack
echo "----------------------------"
echo "Loaded ~/.bashrc"
echo ""
echo "To edit run: bashedit"
echo "To refresh run: bashrefresh"
echo "All aliases: alias"
echo "----------------------------"
printf $NONE
```

and .bash_profile  looks like this (I left it as is because it appeared to already be doing what your code did, but not positive): 

```
test -f ~/.bashrc && source ~/.bashrc
```

**Comment by cgloge** [comment]
It might be that the terminal is getting an incorrect window size.

Try typing in the following to make sure `checkwinsize` is set to `on`
``` 
shopt | grep checkwinsize
``` 

If it's not on, you can turn it on by entering:
```
 shopt -s checkwinsize
``` 

Hope that helps!

**Comment by terichadbourne** [comment]
Thanks for the tip @cgloge. I'd tried that and it's already set to `on`, so not sure what else might be causing it.

**Comment by susanBuck** [comment]
Hi @terichadbourne - I've had that same text wrapping problem, and I've been procrastinating looking into what was causing it. I always assumed it was something with Terminal itself, never thought to look at the .bashrc config as the root of the problem until your post.

Did some digging, and the problem was with how the colors were set up in the .bashrc template I shared. Ref: [OS X Terminal text stacking on top of itself](https://stackoverflow.com/questions/11831296/os-x-terminal-text-stacking-on-top-of-itself).

I've updated the [.bashrc template](https://gist.github.com/susanBuck/ee3a0a53d72198c1a244) to fix the problem, and tested it on my end.

Can you test on your end to confirm?

Thanks for drawing this to my attention!

**Comment by terichadbourne** [comment]
That did the trick for me @susanBuck. Thanks for the detective work! 

Do you need to also change `printf $BBlack` to `printf $Black` since the bold colors are no longer defined? For me it worked (didn't break anything) with  `printf $BBlack` but wasn't bold so I assume it just reverted to its default color and used that.

**Comment by susanBuck** [comment]
Glad that worked!

You're right - it should change to $Black. Will fix that now.

I removed the bold colors just to simplify things.

**Comment by susanBuck** [comment]
It also looks like I've got some stray brackets showing up in the MOTD:

```
You have new mail.
\[\]----------------------------
Loaded ~/.bashrc

To edit run: bashedit
To refresh run: bashrefresh
All aliases: alias
----------------------------
```

Tracking those down now and will follow-up when the template is updated/corrected again.

**Comment by susanBuck** [comment]
Ok, [.bashrc](https://gist.github.com/susanBuck/ee3a0a53d72198c1a244) template is updated again.

I got rid of any color coding on the MOTD. This way it will just use your default foreground text color, making it neutral to whether you use a black or white background in Terminal.

I also updated the function used to determine whether there was changes in a git directory, so be sure to include that when updating your .bashrc file.
